windows/test-app.mjs

```diff
@@ -0,0 +1,228 @@
+        if (fs.existsSync(candidatePath)) {
+            return path.relative('', candidatePath);
+        }
+        //Get parent folder
+        basePath = path.dirname(basePath)
+    }
+
+    return null
+}
+
+function walk(current) {
+    if (!fs.lstatSync(current).isDirectory()) {
+        return [current];
+    }
+
+    const files = fs
+        .readdirSync(current)
+        .map(child => walk(path.join(current, child)));
+    const result = [];
+    return [current, ...files.flat()];
+}
+
+/**
+ * Get a source file and replace parts of its contents.
+ * @param srcPath Path to the source file.
+ * @param replacements e.g. {'TextToBeReplaced': 'Replacement'}
+ * @return The contents of the file with the replacements applied.
+ */
+export function resolveContents(
+    srcPath,
+    replacements
+) {
+    let content = fs.readFileSync(srcPath, 'utf8');
+
+    return Object.keys(replacements).reduce((_, regex) => {
+        return _.replace(new RegExp(regex, 'g'), replacements[regex]);
+    }, content);
+}
+
+//Binary files in React Native Test App Windows project
+const binaryExtensions = ['.png', '.pfx'];
+
+/**
+ * Copy a file to given destination, replacing parts of its contents.
+ * @param srcPath Path to a file to be copied.
+ * @param destPath Destination path.
+ * @param replacements: e.g. {'TextToBeReplaced': 'Replacement'}
+ */
+function copyAndReplace(
+    srcPath,
+    destPath,
+    relativeDestPath,
+    replacements = {}
+) {
+    const fullDestPath = path.join(destPath, relativeDestPath);
+    if (fs.lstatSync(srcPath).isDirectory()) {
+        if (!fs.existsSync(fullDestPath)) {
+            fs.mkdirSync(fullDestPath);
+        }
+        return;
+    }
+
+    const extension = path.extname(srcPath);
+    if (binaryExtensions.indexOf(extension) !== -1) {
+        // Binary file
+        fs.copyFile(srcPath, fullDestPath, err => {
+            if (err) {
+                throw err;
+            }
+        });
+    } else {
+        // Text file
+        const srcPermissions = fs.statSync(srcPath).mode;
+        let content = resolveContents(srcPath, replacements);
+        fs.writeFile(fullDestPath, content, {
+            encoding: 'utf8',
+            mode: srcPermissions,
+        }, err => {
+            if (err) {
+                throw err;
+            }
+        });
+    }
+}
+
+export function copyAndReplaceAll(
+    srcPath,
+    destPath,
+    relativeDestDir,
+    replacements = {}
+) {
+    return Promise.all(walk(srcPath).map((absoluteSrcFilePath) => {
+        const filename = path.relative(srcPath, absoluteSrcFilePath);
+        const relativeDestPath = path.join(relativeDestDir, filename);
+        return copyAndReplace(
+            absoluteSrcFilePath,
+            destPath,
+            relativeDestPath,
+            replacements
+        );
+    }));
+}
+
+const cppSourceFilesExtensions = ['.h', '.cpp', '.idl', '.xaml'];
+
+export function linkSourceFiles(
```

Comment:
Linking source files, so we can continue developing from example app.

Reply:
Do we have to create links? Can't we reference them from `.vcxproj`?

Reply:
We don't need to create links if it's convenient to edit source files from `react-native-test-app/windows` folder while developing. If we want to have both project files and source files in the same folder for development purposes, then we should create links.